robot/packages/cloud_package_manager.go

```diff
@@ -129,7 +129,15 @@ func (m *cloudManager) Sync(ctx context.Context, packages []config.PackageConfig
 
 	// Updated managed map with existingPackages
 	for _, p := range existingPackages {
-		p := p
+		statusFile, err := readStatusFile(p, m.packagesDir)
+		if err != nil {
+			m.logger.Errorf("Failed reading status file for synced package %s: %v", p.Name, err)
+			return multierr.Append(outErr, err)
+		}
+		if statusFile.Status != syncStatusDone {
+			m.logger.Errorf("Package %s was marked as synced but is not in done state", p.Name)
```

Comment:
we can probably check specifically on `syncStatusFailed` and just log something like `Package %s was fully downloaded but failed to unzip, please try a different version` - the current log is confusing for users